azurelinuxagent/common/utils/archive.py

```diff
@@ -38,110 +38,39 @@
 
 ARCHIVE_DIRECTORY_NAME = 'history'
 
-_MAX_ARCHIVED_STATES = 50
+_MAX_ARCHIVED_STATES = 100
```

Comment:
multiplying by 2, since now we also have an archive for the agent status per goal state